iroh/src/client/blobs.rs

```diff
@@ -368,6 +390,263 @@ where
     }
 }
 
+/// A scope in which blobs can be added.
+#[derive(derive_more::Debug)]
+struct BatchInner<C: ServiceConnection<RpcService>> {
+    /// The id of the scope.
+    scope: u64,
+    /// The rpc client.
+    rpc: RpcClient<RpcService, C>,
+    /// The stream to send drop
+    #[debug(skip)]
+    updates: Mutex<UpdateSink<RpcService, C, BatchUpdate>>,
+}
+
+/// A batch for write operations.
+///
+/// This serves mostly as a scope for temporary tags.
+///
+/// It is not a transaction, so things in a batch are not atomic. Also, there is
+/// no isolation between batches.
+#[derive(derive_more::Debug)]
+pub struct Batch<C: ServiceConnection<RpcService>>(Arc<BatchInner<C>>);
```

Comment:
Is there no `commit` step or at least `close` step necessary?

Reply:
No. A batch is not a transaction. So as long as the batch lives, the temp tags are alive. Closing is just on drop. Note that this is a bit sloppy, so in some cases like network error it will take some time for the store to notice that a batch is gone. But that is fine in the context of temp tags, it just means that the data will hang around a bit longer before being purged by the next gc.

Reply:
I think it would be good to add this to the comments in the code

Reply:
There is a comment on the Batch struct. Maybe move it to the fn batch()?